File: app/src/main/java/com/codemybrainsout/imageviewer/view/photo/PhotoContract.java
package com.codemybrainsout.imageviewer.view.photo;

import com.codemybrainsout.imageviewer.model.Photo;
import com.codemybrainsout.imageviewer.model.User;
import com.codemybrainsout.imageviewer.view.base.BasePresenter;
import com.codemybrainsout.imageviewer.view.base.BaseView;

import java.util.List;

/**
 * Created by ahulr on 09-06-2017.
 */

public class PhotoContract {

    interface Presenter extends BasePresenter {

        void openUser(User user);

        void openPhoto(Photo photo);

        void loadPhotos(int page, int limit, String orderBy);

        void loadMorePhotos(int page, int limit, String orderBy);
    }

    interface View extends BaseView<Presenter> {

        void showUser(User user);

        void showPhoto(Photo photo);

        void refreshPhotos(List<Photo> list);

        void addPhotos(List<Photo> list);

    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/photo/PhotoContract.kt
package com.codemybrainsout.imageviewer.view.photo

import com.codemybrainsout.imageviewer.view.base.BasePresenter
import com.codemybrainsout.imageviewer.view.base.BaseView
import com.codemybrainsout.imageviewer.model.*

/**
 * Created by ahulr on 09-06-2017.
 */
class PhotoContract {
    interface Presenter : BasePresenter {
        fun openUser(user: User?)
        fun openPhoto(photo: Photo?)
        fun loadPhotos(page: Int, limit: Int, orderBy: String?)
        fun loadMorePhotos(page: Int, limit: Int, orderBy: String?)
    }

    interface View : BaseView<Presenter?> {
        fun showUser(user: User?)
        fun showPhoto(photo: Photo?)
        fun refreshPhotos(list: List<Photo>)
        fun addPhotos(list: List<Photo>)
    }
}